src/viewProviders/RokuAutomationViewViewProvider.ts

```diff
@@ -47,6 +48,85 @@ export class RokuAutomationViewViewProvider extends BaseRdbViewProvider {
             }
         });
 
+        this.registerCommand(VscodeCommand.rokuAutomationViewLoadAutomation, async () => {
+            if (this.isRecording) {
+                // Only allow loading when we aren't currently recording
+                await this.setIsRecording(false);
+            }
+
+            // macOS does not have a title bar on the open dialog so we need to show a warning message
+            if (process.platform === 'darwin') {
+                const confirm = await vscode.window.showWarningMessage(
+                    'This will replace all currently loaded automation scripts. Continue?',
+                    { modal: true },
+                    'Yes'
+                );
+                if (confirm !== 'Yes') {
+                    return;
+                }
+            }
+
+            vscode.window.showOpenDialog({
+                title: 'Load Automation Scripts (Warning: This will replace all currently loaded scripts)',
+                filters: {
+                    'JSON': ['json'],
+                    'All Files': ['*']
+                },
+                defaultUri: vscode.Uri.file('automation.json'),
+                canSelectMany: false
+            }).then(async (filePath) => {
+                if (!filePath) {
+                    return;
+                }
+                try {
+                    const data = readFileSync(filePath[0].fsPath, 'utf8');
+                    const result = JSON.parse(data);
+                    this.selectedConfig = result.selectedConfig;
+                    this.rokuAutomationConfigs = result.configs;
+                    await this.extensionContext.workspaceState.update(WorkspaceStateKey.rokuAutomationConfigs, JSON.stringify(result));
+                    const message = this.createEventMessage(ViewProviderEvent.onRokuAutomationConfigsLoaded, {
+                        selectedConfig: this.selectedConfig,
+                        configs: this.rokuAutomationConfigs
+                    });
+
+                    this.postOrQueueMessage(message);
+
+                    this.updateCurrentRunningStep();
+                    this.onLoadAutomation();
+                    vscode.window.showInformationMessage('Automation loaded successfully from ' + filePath[0].fsPath);
+                } catch (err) {
+                    vscode.window.showErrorMessage('Failed to load automation: ' + (err as Error).message);
+                }
+            });
+        });
+
+        this.registerCommand(VscodeCommand.rokuAutomationViewExportAutomation, async () => {
+            vscode.window.showInformationMessage('Exporting automation data...');
+
+            if (this.isRecording) {
+                // Only allow exporting when we aren't currently recording
+                await this.setIsRecording(false);
+            }
+            vscode.window.showSaveDialog({
```

Comment:
Optional: You could write this as `const filePath = await vscode.window.showSaveDialog({` to make the code more readable